### DISCUSSION TITLE
Executing tfc_end2end_example

### DISCUSSION THREAD

**Discussion by btwbtw01** [Xilinx/finn#503] [Q&A] [answered]
Hi, I want to ask can we train QNN with Brevitas outside the docker? then put our model into FINN compiler to form hardware? If the answer is yes, Do we need to install onnx? Is onnx already in FINN docker?
![image](https://user-images.githubusercontent.com/85877169/150144667-66ebaf4a-b5f6-4f7b-a788-dd1fe5f3e71f.png)

![image](https://user-images.githubusercontent.com/85877169/150144637-ef289ed5-25cb-416b-b729-14cbf1e8377b.png)

Can someone help me?

**Comment by nhma20** [comment, marked as answer]
You can define and train your network with Brevitas outside the FINN docker environment if you have Brevitas and its dependencies installed. For ONNX, I believe you need both the onnx and onnxoptimizer packages which can be installed with '''pip install onnx" and "pip install onnxoptimizer" respectively. They should already be in the FINN docker.

**Reply by btwbtw01** [reply]
Thanks for your answer. I will try it, and I already run example in the docker successfully except compiling
